internal/backend/mqtt/backend.go

```diff
@@ -116,6 +116,8 @@ func NewBackend(config BackendConfig) (*Backend, error) {
 		return nil, fmt.Errorf("mqtt: unknown auth type: %s", config.Auth.Type)
 	}
 
+	b.qos = config.Auth.Generic.QOS
```

Comment:
I don't know if it would be better to have this in the Backend instantiation at line 87. Feel free to amend.